### PR TITLE
[flang][Driver] Add -masm option to flang

### PR DESCRIPTION
**Pull request by DavidSpickett**
The motivation here was a suggestion over in Compiler Explorer. You can use `-mllvm` already to do this but since gfortran supports `-masm`, I figured I'd try to add it.

This is done by flang expanding `-masm` into `-mllvm x86-asm-syntax=`, then passing that to fc1. Which then collects all the `-mllvm` options and forwards them on.

The code to expand it comes from clang `Clang::AddX86TargetArgs` (there are some other places doing the same thing too). However I've removed the `-inline-asm` that clang adds, as fortran doesn't have inline assembly.

So `-masm` for flang purely changes the style of assembly output.

```
$ ./bin/flang-new /tmp/test.f90 -o - -S -target x86_64-linux-gnu
<...>
        pushq   %rbp
$ ./bin/flang-new /tmp/test.f90 -o - -S -target x86_64-linux-gnu -masm=att
<...>
        pushq   %rbp
$ ./bin/flang-new /tmp/test.f90 -o - -S -target x86_64-linux-gnu -masm=intel
<...>
        push    rbp
```

The test is adapted from `clang/test/Driver/masm.c` by removing the clang-cl related lines and changing the 32 bit triples to 64 bit triples since flang doesn't support 32 bit targets.

### CODE REVIEW

File: flang/test/Driver/masm.f90 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// RUN: %flang -target x86_64-unknown-linux -masm=intel -S %s -### 2>&1 | FileCheck --check-prefix=CHECK-INTEL %s
```
Comment (banach-space) [inline flang/test/Driver/masm.f90:1]: 1. Use Fortran style comments :)
2. Missing requires?

**Review by banach-space — APPROVED** [review]
Nice, thank you!

File: flang/test/Driver/masm.f90 (hunk 1 above)
Comment (MaskRay) [reply flang/test/Driver/masm.f90:1]: `--target=` for new tests

Comment (DavidSpickett) [reply flang/test/Driver/masm.f90]: 1. Done, I forgot this wasn't actually parsing the file. So I've removed the function as well, it's not needed.
2. Don't need requires if all we're doing is using the triple name, I checked that by building without x86 and the test still works.
3. Used --target= for the run lines.

**Pull request merged**